### PR TITLE
Adding nxos_vxlan_vtep_vni

### PR DESCRIPTION
**Pull request by GGabriele**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

nxos_vxlan_vtep_vni
##### ANSIBLE VERSION

```
ansible 2.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/usr/share/ansible/']
```
##### SUMMARY

Creates a Virtual Network Identifier member (VNI)

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks again to @GGabriele. This module is going into community review.

We notice that you are one of the maintainers (@privateip, @GGabriele, @gundalow) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

File: network/nxos/nxos_vxlan_vtep_vni.py [hunk 1]
```diff
@@ -0,0 +1,566 @@
...
+        commands = ['no member vni {0} associate-vrf'.format(
+                                                module.params['vni'])]
+    else:
+        commands = ['no member vni {0}'.format(module.params['vni'])]
+    parents = ['interface {0}'.format(module.params['interface'])]
+    candidate.add(commands, parents=parents)
+
+
+def main():
+    argument_spec = dict(
```
Comment (gundalow) [inline network/nxos/nxos_vxlan_vtep_vni.py:480]: There are more options defined here than in the `DOCUMENTATION:`

Any option specified in `extends_documentation_fragment: nxos` needs defining in `module_utils`, have a look at other network modules.

File: network/nxos/nxos_vxlan_vtep_vni.py [hunk 2]
```diff
@@ -0,0 +1,566 @@
...
+                       required=False),
+            include_defaults=dict(default=True),
+            config=dict(),
+            save=dict(type='bool', default=False)
+    )
+    module = get_network_module(argument_spec=argument_spec,
+                                supports_check_mode=True)
+
+    if module.params['assoc_vrf']:
+        mutually_exclusive_params = ['multicast_group',
```
Comment (gundalow) [inline network/nxos/nxos_vxlan_vtep_vni.py]: Please look at examples of `mutually_exclusive`, this is built in validation,  e.g cloud/openstack/_keystone_user.py

**Comment by gundalow** [comment]
needs_revision

**Comment by gregdek** [comment]
Thanks @GGabriele for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by GGabriele** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @GGabriele. To the current maintainers, @jedelman8, @GGabriele, @privateip, @gundalow please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gundalow** [comment]
@GGabriele I may be missing it, but where is the documentation defined for the following options?

```
m_facts
state
include_defaults
config
save
```

I don't see them defined in `module_docs_fragment/nxos.py`, nor do I see an PR in flight at https://ansible.sivel.net/pr/byfile.html

Apart from this issue privateip and I are happy

needs_revision

**Comment by gregdek** [comment]
Thanks @GGabriele for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by GGabriele** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @GGabriele. To the current maintainers, @jedelman8, @GGabriele, @privateip, @gundalow please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: network/nxos/nxos_vxlan_vtep_vni.py [hunk 3]
```diff
@@ -78,6 +78,26 @@
         required: false
         default: present
         choices: ['present','absent']
+    include_defaults:
+        description:
+            - Specify to use or not the complete runnning configuration
```
Comment (gundalow) [inline network/nxos/nxos_vxlan_vtep_vni.py:83]: s/runnning/running/

File: network/nxos/nxos_vxlan_vtep_vni.py [hunk 4]
```diff
@@ -78,6 +78,26 @@
         required: false
         default: present
         choices: ['present','absent']
+    include_defaults:
```
Comment (gundalow) [inline network/nxos/nxos_vxlan_vtep_vni.py:81]: @privateip Are we standardising on `defaults:` or `include_defaults:` (Assuming this is the same thing)

**Review by gundalow — CHANGES_REQUESTED** [review]
needs_revision

**Pull request merged**